BitList.cs

```diff
@@ -0,0 +1,1553 @@
+    {
+        var _bytesList = new List<byte[]>();
+        var _byteList = new List<byte>();
+        foreach (var str in bits)
+        {
+            var bytesList = new List<byte[]>();
+            foreach (var chr in str)
+                bytesList.Add(BitConverter.GetBytes(chr));
+            var byteList = new List<byte>();
+            foreach (var bytes in bytesList)
+            {
+                foreach (var num in bytes)
+                    byteList.Add(num);
+            }
+            _bytesList.Add(byteList.ToArray());
+        }
+        foreach (var bytes in _bytesList)
+        {
+            foreach (var num in bytes)
+                _byteList.Add(num);
+        }
+        this.bits = new BitArray(_byteList.ToArray());
+    }
+
+    IEnumerator<bool> IEnumerable<bool>.GetEnumerator()
+    {
+        return new BitListEnumerator(this);
+    }
+
+    IEnumerator IEnumerable.GetEnumerator()
+    {
+        return new BitListEnumerator(this);
+    }
+
+    /// <summary>
+    /// Get the index of a bit in this instance.
+    /// </summary>
+    /// <param name="bit">A bit.</param>
+    /// <returns>The <paramref name="item"/> index.</returns>
+    public int IndexOf(bool item)
+    {
+        for (int i = 0; i < Count; i++)
+        {
+            if (item == this[i])
+                return i;
+        }
+        return -1;
+    }
+
+    /// <summary>
+    /// Get the index of a list of bit in this instance.
+    /// </summary>
+    /// <param name="items">A list of bit.</param>
+    public int IndexOf(BitList items)
+    {
+        for (int i = 0; i < Count; i++)
+        {
+            if (items == this[i, bits.Count])
+                return i;
+        }
+        return -1;
+    }
+
+    /// <summary>
+    /// Clone this instance in a new object.
+    /// </summary>
+    public object Clone()
+    {
+        return this;
+    }
+
+    /// <summary>
+    /// Return true if the specified BitList is equal to this istance.
+    /// </summary>
+    /// <param name="other">A list of bit.</param>
+    public bool Equals(BitList other)
+    {
+        return Equals(this, other);
+    }
+
+    /// <summary>
+    /// Return true if this instance contains the specified bit.
+    /// </summary>
+    /// <param name="item">A bit.</param>
+    public bool Contains(bool item)
+    {
+        return IndexOf(item) != -1;
+    }
+
+    /// <summary>
+    /// Return true if this instance contains the specified list of bit.
+    /// </summary>
+    /// <param name="items">A list of bit.</param>
+    public bool Contains(BitList items)
+    {
+        return IndexOf(items) != -1;
+    }
+
+    /// <summary>
+    /// Shift bits to left.
+    /// </summary>
+    public void LeftShift(int count)
+    {
+        if (count != 0)
+        {
+            for (int i = 1; i == Math.Abs(count); i++)
+            {
+                if (count > 0)
+                    Add(false);
+                else
+                {
+                    try
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)